coral/src/app/features/topics/request/TopicRequest.test.tsx

```diff
@@ -663,4 +681,135 @@ describe("<TopicRequest />", () => {
+        },
+      });
+
+      mockgetTopicAdvancedConfigOptions({
+        mswInstance: server,
+        response: {
+          data: defaultgetTopicAdvancedConfigOptionsResponse,
+        },
+      });
+
+      customRender(
+        <AquariumContext>
+          <TopicRequest />
+        </AquariumContext>,
+        { queryClient: true }
+      );
+
+      // Wait all API calls to resolve, which are required for the render
+      await screen.findByLabelText("Environment");
+      await screen.findByRole("option", { name: "DEV" });
+    });
+
+    afterAll(() => {
+      cleanup();
+    });
+
+    beforeEach(async () => {
+      // Fill form with valid data
+      await user.selectOptions(screen.getByLabelText("Environment"), "DEV");
+      await user.clear(screen.getByLabelText("Topic name*"));
+      await user.clear(screen.getByLabelText("Topic partitions*"));
+      await user.clear(screen.getByLabelText("Replication factor*"));
+      await user.clear(screen.getByLabelText("Description*"));
+
+      await user.type(
+        screen.getByLabelText("Topic name*"),
+        "this-is-topic-name{tab}"
+      );
+      await user.type(screen.getByLabelText("Topic partitions*"), "3{tab}");
+      await user.type(screen.getByLabelText("Replication factor*"), "3{tab}");
+      await user.type(screen.getByLabelText("Description*"), "test{tab}");
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    describe("when API returns an error", () => {
+      beforeEach(async () => {
+        mockRequestTopic({
+          mswInstance: server,
+          response: {
+            data: { message: "Topic with such name already exists!" },
+            status: 400,
+          },
+        });
+      });
+
+      it("renders an error message", async () => {
+        const spyPost = jest.spyOn(api, "post");
+
+        await user.click(screen.getByRole("button", { name: "Request topic" }));
+
+        await waitFor(() => {
+          const btn = screen.getByRole("button", { name: "Request topic" });
+          expect(btn).toBeDisabled();
+        });
+
+        expect(spyPost).toHaveBeenCalledTimes(1);
+        expect(spyPost).toHaveBeenCalledWith("/createTopics", {
+          environment: "1",
+          topicname: "this-is-topic-name",
+          replicationfactor: "3",
+          topicpartitions: 3,
+          advancedTopicConfigEntries: [],
+          description: "test",
+          remarks: "",
+          topictype: "Create",
+        });
+
+        const alert = await screen.findByRole("alert");
+        expect(alert).toHaveTextContent("Topic with such name already exists!");
+      });
+    });
+    describe("when API request is successful", () => {
+      beforeEach(async () => {
+        mockRequestTopic({
+          mswInstance: server,
+          response: { data: { status: "200 OK" } },
+        });
+      });
+
+      it("it redirects user to myTopicRequests", async () => {
```

Comment:
Minor comment - the "it" in the 'it' is redundant (nothing blocking tho, just to raise this)